scripts/turretimpulse.lua

```diff
@@ -0,0 +1,132 @@
+include "constants.lua"
+local WOBBLE_DIST = 2
+local WOBBLE_SPEED = 2
+local TURRET_AIM_SPEED = math.rad(300)
+local TURRET_SPIN_SPEED = math.rad(60)
+local RING_SPIN_SPEED = -1 * math.rad(120)
+local CRYSTAL_TURN_SPEED = math.rad(150)
+
+
+-- pieces --
+local gp = piece('gp')
+local base = piece('base')
+local turret = piece('turret')
+local ring = piece('ring')
+local center = piece('center')
+local firepoint = piece('firepoint')
+local crystals = {}
+local state = true
+crystals[1] = piece('crystal1')
+crystals[2] = piece('crystal2')
+crystals[3] = piece('crystal3')
+crystals[4] = piece('crystal4')
+
+local smokePiece = {base}
+
+local function WobbleTurret()
+	while true do
+		Move(turret, y_axis, WOBBLE_DIST, WOBBLE_SPEED)
+		WaitForMove(turret, y_axis)
+		Move(turret, y_axis, -1 * WOBBLE_DIST, WOBBLE_SPEED)
+		WaitForMove(turret, y_axis)
+		Sleep(100)
```

Comment:
no sleep in the original
```suggestion
		Move(turret, y_axis, -WOBBLE_DIST, WOBBLE_SPEED)
		WaitForMove(turret, y_axis)
```

Reply:
This may cause hangs. I received a hang during initial testing when a newton was created (may or may not be false attribution) so I included it to be safe.